include/VecCore/VecMath.h

```diff
@@ -80,23 +64,116 @@ T Sign(const T &x)
   return CopySign(T(1), x);
 }
 
-// Trigonometric Functions
+#ifdef VECCORE_ENABLE_LIBMVEC
+
+#define VECMATH_REAL_FUNC2(f, name)                          \
+  template <typename T>                                      \
+  VECCORE_FORCE_INLINE                                       \
+  VECCORE_ATT_HOST_DEVICE                                    \
+  T f(const T &x, const T &y)                                \
+  {                                                          \
+      T output;                                              \
+      int v_size = vecCore::VectorSize(x);                   \
+                                                             \
+      #pragma omp simd                                       \
+      for(int i = 0; i < v_size; i++)                        \
+      {                                                      \
+        Scalar<T> input1 = vecCore::Get(x, i)                \
+        Scalar<T> input2 = vecCore::Get(y, i)                \
+        vecCore::Set(output, i, std::name(input1, input2));  \
+      }                                                      \
+    return output;                                           \
+  }
 
-template <typename T>
-VECCORE_FORCE_INLINE
-VECCORE_ATT_HOST_DEVICE
-T Sin(const T &x)
-{
-  return std::sin(x);
-}
+#else
 
-template <typename T>
-VECCORE_FORCE_INLINE
-VECCORE_ATT_HOST_DEVICE
-T Cos(const T &x)
-{
-  return std::cos(x);
-}
+# define VECMATH_REAL_FUNC2(f, name)                         \
+  template <typename T>                                      \
+  VECCORE_FORCE_INLINE                                       \
+  VECCORE_ATT_HOST_DEVICE                                    \
+  T f(const T &x, const T &y)                                \
+  {                                                          \
+      return std::name(x, y);                                \
+  }
+
+#endif
+
+VECMATH_REAL_FUNC2(CopySign, copysign)
+VECMATH_REAL_FUNC2(ATan2, atan2)
+VECMATH_REAL_FUNC2(Pow, pow)
+VECMATH_REAL_FUNC2(Hypot, hypot)
+VECMATH_REAL_FUNC2(Fmod, fmod)
+
+#undef VECMATH_REAL_FUNC2
+
+
+#ifdef VECCORE_ENABLE_LIBMVEC
```

Comment:
To check validity, if we change this to `ifndef` then the same error `'#' is not followed by a macro parameter` comes up again.

Reply:
Corrected with using `_Pragma` now

Reply:
What compilers is _Pragma defined in? (i.e. is it guaranteed to be there)?

Reply:
When the tests were run by Travis, it worked fine with Clang. This specific pragma is defined for GCC 4.9 and above, so GCC 4.8 which is currently used in this repo ignored it.
It exists for ICC [as well](https://software.intel.com/en-us/forums/intel-c-compiler/topic/545048).

Reply:
The problem is not with the `_Pragma`. If it's supported by GCC, Clang, and ICC, it's more than fine. The only problem is that with the loop the functions become slower than they currently are. So it's better to add all benchmarks first to make sure we don't introduce regressions, then make the modifications to the functions.